source/posts/good_great_10x.markdown

```diff
@@ -0,0 +1,63 @@
+
+### "Not being afraid to throw away code once you understand the problem better."
+
+I once read that the moment you compile code and deploy it, it becomes legacy code. This means it’s a responsibility at best and a liability at worst. The trouble is that we become attached to things...even code.
+
+For a long time, I saw my code as something that people would look at and associate with me. When the opportunity or need to throw away my code came along, especially if somebody else suggested it...I’d take it as a personal attack. I’ve heard [@orta](https://twitter.com/orta) respond to this with, *“You are not your code.”* It’s a simple reminder that we are all growing and learning and our industry is still changing. It’s something you can appreciate whenever you look at code you wrote a month ago, or a year ago, or 10 years ago.
+
+Getting rid of your code is not personal. We all did the best we could when we wrote the first version of something. Don’t be afraid to have “throwing it all out” be one of your steps for iterating on a design.
+
+### "One who uses the proper metaphor, and strives to find the proper metaphor."
+
+This is great. It's a larger version of the problem I often run into of spending a rather large amount of time thinking about what to name a variable.
+
+When you spend the time to find or create the right metaphor for something you are building, you don’t have to explain it so much to those who will use it.
+
+Years ago, [Metaphors We Live By](http://www.amazon.com/Metaphors-We-Live-George-Lakoff/dp/0226468011) was recommended to me by [Venkatesh Rao](http://ribbonfarm.com). If you want more insight into how we use metaphors to share concepts with others (which is pretty important in programming) and how metaphor affects both our thinking and our language...I highly recommend reading this. In fact, I’ll be re-reading it soon.
+
+One problem I’ve seen is when somebody uses a metaphor that is **almost** appropriate for what they are building. If you have a class or variable that is essentially a `Foo` except for two common edge cases where it behaves like a `Bar` you will inevitably have somebody who uses it exclusively like a `Foo` and causes problems.
+
+### What about the 10x programmer?
+
+Many have written supporting or denying the existence of the 10x programmer. I’m squarely in the ‘they do exist’ camp, but perhaps not in the traditional sense.
+
+I recently came across (where was this!!!) a mention that *“the 10x programmer is one who makes the 5 closest programmers twice as effective.”*
```

Comment:
That's a strong concept. `Enhancing the field in general` is one of those phrases that is beautifully open for interpretation.

If some master of technique and subject matter came along and created a pod called `Monad` that clarified and conveyed monads in a succinct way, that would enhance the field.

I'll see if I can work this in.